subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleDependency.java

```diff
@@ -221,13 +223,26 @@ public interface ModuleDependency extends Dependency, HasConfigurableAttributes<
     ModuleDependency capabilities(Action<? super ModuleDependencyCapabilitiesHandler> configureAction);
 
     /**
-     * Returns the set of requested capabilities for this dependency.
-     * @return An immutable view of requested capabilities. Updates must be done calling {@link #capabilities(Action)}.
+     * Returns the set of explicitly requested capabilities for this dependency.
```

Comment:
This should be deprecated eventually. 

Looking at AGP and KGP sources, it actually does not seem like either use this method so we might be able to do this faster than I had presumed.

However, we cannot deprecate this without de-incubating `getCapabilitySelectors` so for the time being we'll likely need to leave this in. 

I've written an issue: https://github.com/gradle/gradle/issues/30528

It is listed as team-triage. We can determine if we want to immediately de-incubate this API and fast-track the deprecation for 9.0, but that would mean we need to decide quickly so we can deprecate in 8.11

In any case, I think we should leave that decision for a follow up.

Reply:
As for returning a list. Yes it does, but that list is derived from a set. There are no duplicates in the list.

We can say list here but I think I prefer saying `set` since the returned value does satisfy set semantics

Reply:
Can we say "returns a list (that contains the set of...)" then?  It just looks like a typo to me, and makes the rest of the comment appear untrustworthy.

Reply:
Rewritten to a form that does not contain 'set' or 'list'